Search/Factory.php

```diff
@@ -49,13 +53,37 @@ public function makeField($name, $value, $type = Field::TYPE_STRING)
     }
 
     /**
-     * Make a new metadata object representing a mapped
-     * searchable class
+     * Make a new metadata object repesenting a mapped
+     * searchanle class
+     */
+    public function makeClassMetadata($class)
+    {
+        return new ClassMetadata($class);
+    }
+
+    /**
+     * Make a new metadata object representing an index for
+     * a mapped searchable class.
      *
      * @return IndexMetadata
      */
-    public function makeIndexMetadata($class)
+    public function makeIndexMetadata()
+    {
+        return new IndexMetadata();
+    }
+
+    public function makeMetadataField($name)
+    {
+        return new MetadataField($name);
+    }
+
+    public function makeMetadataProperty($path)
```

Comment:
Create is more common in that I have seen that used before, I choose `make` after looking at [wikipedia](http://en.wikipedia.org/wiki/Factory_method_pattern). Its got less characters than create and factorys "make" things in the real world. Normally you wouldn't say that a factory "creates" things. So in balance, I prefer "make".